pkg/hooks/read.go

```diff
@@ -80,12 +80,13 @@ func ReadDir(path string, extensionStages []string, hooks map[string]*current.Ho
 					continue
 				}
 			}
-			return err
+			res = err
```

Comment:
This will only report the last error and we'll lose previous ones. Can we instead do as follows?
```golang
if res == nil {
   res = err
} else {
   res = errors.Wrapf(res, "%v", err)
}
```

Reply:
`errors.Wrapf(nil, ...)` always returns nil, so we need to make it conditional.